### ISSUE TITLE
Read and summarize the RFCs 2813, 2812, 2811, 2810, and 1459

### ISSUE THREAD

**Issue by JonathanDUFOUR** [Saphirac/ft_irc#5]
Before starting to implement anything for our IRC server, we should read those RFCs because they are a good reference on how the IRC protocol works and thus how we implement an IRC server. We should read them from the most recent to the less recent.
And as we read them, we should make a brief document to summarize what matters to us.
Therefore, I suggest that we create a MarckDown file at the root of the project for that purpose.
Let's call it `RFCs_summary.md`.

**Comment by alexandre-hallaine** [comment]
Hi there,

I wanted to mention that RFC 1459 is experimental and largely superseded by RFC 2812, so I'd suggest focusing on RFC 2812. It's the most important one in terms of messaging and message structure.

**Key Points in RFC 2812:**

* [Connection Registration (Section 3.1)](https://datatracker.ietf.org/doc/html/rfc2812#section-3.1): Defines messages exchanged during initial connection process.
* [Numeric Replies (Sections 2.4)](https://datatracker.ietf.org/doc/html/rfc2812#section-2.4): Standardized responses from the server to client commands.

I'd also recommend checking out [IRC2Numerics website](https://www.alien.net.au/irc/irc2numerics.html) for comprehensive overview of IRC messages and their corresponding numeric replies.

Have fun coding your IRC server!

**Comment by JonathanDUFOUR** [comment]
After encountering some limitations due to the MarckDown format, it has been decided to switch to the HTML format for the summary of the RFCs.
